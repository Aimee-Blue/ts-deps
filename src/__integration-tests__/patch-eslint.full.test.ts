import {
  buildAndPack,
  ROOT,
  emptyDirSafe,
  spawnAndCheck,
  sortedDirectoryContents,
} from './helpers';
import { join } from 'path';
import { ensureDir, copy } from 'fs-extra';

jest.setTimeout(5 * 60 * 1000);

describe('init', () => {
  describe('given fresh directory after init', () => {
    const initDir = join(ROOT, 'integration-test-init-full-eslint');

    beforeAll(async () => {
      await emptyDirSafe(initDir);
      await ensureDir(initDir);

      // to maky sure husky will not conflict with hooks of the main repo
      spawnAndCheck('git', ['init'], {
        cwd: initDir,
      });

      spawnAndCheck('npm', ['init', '-y'], {
        cwd: initDir,
      });

      const sourceLocation = join(ROOT, 'test-sources', 'init-full');

      await copy(sourceLocation, initDir, {
        overwrite: true,
      });

      spawnAndCheck('npm', ['add', 'ts-deps@0.12.8', '--save-dev'], {
        cwd: initDir,
      });

      spawnAndCheck('npx', ['ts-deps@0.12.8', 'init', '--force'], {
        cwd: initDir,
      });
    });

    it('should init with correct files', async () => {
      const contents = await sortedDirectoryContents(initDir);

      expect(contents).toMatchSnapshot();

      expect(contents).toContain('tslint.json');
      expect(contents).not.toContain('.eslintignore');
      expect(contents).not.toContain('.eslintrc.js');
    });

    describe('when upgraded', () => {
      beforeAll(async () => {
        const { packageLocation } = await buildAndPack();

        spawnAndCheck('npm', ['remove', 'ts-deps', '--save-dev'], {
          cwd: initDir,
        });

        spawnAndCheck('npm', ['add', 'file:' + packageLocation, '--save-dev'], {
          cwd: initDir,
        });

        spawnAndCheck('npm', ['run', 'patch'], {
          cwd: initDir,
        });
      });

      it('should patch correctly', async () => {
        const contents = await sortedDirectoryContents(initDir);

        expect(contents).toMatchSnapshot();
        expect(contents).not.toContain('tslint.json');
        expect(contents).toContain('.eslintignore');
        expect(contents).toContain('.eslintrc.js');
      });

      it('should lint', () => {
        spawnAndCheck('npm', ['run', 'lint'], {
          cwd: initDir,
        });
      });
    });
  });
});
