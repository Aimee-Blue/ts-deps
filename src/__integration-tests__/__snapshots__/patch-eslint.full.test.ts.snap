// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`init given fresh directory after init should init with correct files 1`] = `
Array [
  ".gitignore",
  ".yarnrc",
  "babel.config.js",
  "commitlint.config.js",
  "jest.config.integration.js",
  "jest.config.js",
  "package-lock.json",
  "package.json",
  "prettier.config.js",
  "release.config.js",
  "scripts/",
  "scripts/build.js",
  "scripts/clean.js",
  "scripts/combineCoverage.js",
  "scripts/release.js",
  "src/",
  "src/__integration-tests__/",
  "src/__tests__/",
  "src/__tests__/max.test.ts",
  "src/__tests__/min.test.js",
  "src/__tests__/test.d.ts",
  "src/index.ts",
  "src/max/",
  "src/max/index.d.ts",
  "src/max/index.js",
  "src/max/max.ts",
  "src/min/",
  "src/min/index.d.ts",
  "src/min/index.js",
  "src/min/min.js",
  "src/other.d.ts",
  "src/other.js",
  "ts-deps.config.js",
  "tsconfig.declarations.json",
  "tsconfig.json",
  "tslint.json",
]
`;

exports[`init given fresh directory after init when upgraded should patch correctly 1`] = `
Array [
  ".eslintignore",
  ".eslintrc.js",
  ".gitignore",
  ".yarnrc",
  "babel.config.js",
  "commitlint.config.js",
  "jest.config.integration.js",
  "jest.config.js",
  "package-lock.json",
  "package.json",
  "prettier.config.js",
  "release.config.js",
  "scripts/",
  "scripts/build.js",
  "scripts/clean.js",
  "scripts/combineCoverage.js",
  "scripts/release.js",
  "src/",
  "src/__integration-tests__/",
  "src/__tests__/",
  "src/__tests__/max.test.ts",
  "src/__tests__/min.test.js",
  "src/__tests__/test.d.ts",
  "src/index.ts",
  "src/max/",
  "src/max/index.d.ts",
  "src/max/index.js",
  "src/max/max.ts",
  "src/min/",
  "src/min/index.d.ts",
  "src/min/index.js",
  "src/min/min.js",
  "src/other.d.ts",
  "src/other.js",
  "ts-deps.config.js",
  "tsconfig.declarations.json",
  "tsconfig.json",
]
`;
